pkg/tfvars/baremetal/baremetal_test.go

```diff
@@ -0,0 +1,105 @@
+// Package baremetal contains bare metal specific Terraform-variable logic.
+package baremetal
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/openshift/installer/pkg/types/baremetal"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFilterWorkers(t *testing.T) {
+
+	cases := []struct {
+		scenario string
+
+		libvirtURI         string
+		apiVIP             string
+		imageCacheIP       string
+		bootstrapOSImage   string
+		externalBridge     string
+		externalMAC        string
+		provisioningBridge string
+		provisioningMAC    string
+		platformHosts      []*baremetal.Host
+		image              string
+		ironicUsername     string
+		ironicPassword     string
+		ignition           string
+
+		expectedError string
+		expectedHosts []string
+	}{
+		{
+			scenario: "filter-workers",
+
+			platformHosts: []*baremetal.Host{
+				{
+					Name:            "master-0",
+					Role:            "master",
+					HardwareProfile: "default",
+					BMC: baremetal.BMC{
+						Address: "redfish+http://192.168.111.1:8000/redfish/v1/Systems/e4427260-6250-4df9-9e8a-120f78a46aa6",
+					},
+				},
+				{
+					Name:            "master-1",
+					Role:            "master",
+					HardwareProfile: "default",
+					BMC: baremetal.BMC{
+						Address: "redfish+http://192.168.111.1:8000/redfish/v1/Systems/e4427260-6250-4df9-9e8a-120f78a46aa6",
+					},
+				},
+				{
+					Name:            "worker-0",
+					Role:            "worker",
+					HardwareProfile: "default",
+					BMC: baremetal.BMC{
+						Address: "redfish+http://192.168.111.1:8000/redfish/v1/Systems/5ff80e96-f131-4717-b85e-5b045ba7595e",
+					},
+				},
+			},
+
+			expectedHosts: []string{"master-0", "master-1"},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.scenario, func(t *testing.T) {
+
+			imageDownloader = func(baseURL string) (string, error) {
+				return "", nil
+			}
+
+			data, err := TFVars(
+				tc.libvirtURI,
+				tc.apiVIP,
+				tc.imageCacheIP,
+				tc.bootstrapOSImage,
+				tc.externalBridge,
+				tc.externalMAC,
+				tc.provisioningBridge,
+				tc.provisioningMAC,
+				tc.platformHosts,
+				tc.image,
+				tc.ironicUsername,
+				tc.ironicPassword,
+				tc.ignition)
+
+			if tc.expectedError == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.Regexp(t, tc.expectedError, err)
+			}
+
+			var cfg config
+			err = json.Unmarshal(data, &cfg)
+			assert.Nil(t, err)
+
+			for i, hostName := range tc.expectedHosts {
+				assert.Equal(t, hostName, cfg.Masters[i]["name"])
+			}
+		})
+	}
+}
```

Comment:
Do we need a test case here where the masters and workers are out of order?

Reply:
Good idea